lib/kernels/clink_kernels.cc

```diff
@@ -64,19 +52,38 @@ AsyncValueRef<double> SquareAdd(Argument<double> x, Argument<double> y,
 
 double Square(double x) { return x * x; }
 
-void OneHotEncoderLoad(Argument<std::string> path,
-                       Result<RCReference<OneHotEncoderModel>> result_model,
-                       KernelErrorHandler handler,
-                       const ExecutionContext &exec_ctx) {
-  auto model = OneHotEncoderModel::load(path.get(), exec_ctx.host());
-  CLINK_RETURN_IF_ERROR(handler, model.takeError());
-  result_model.Emplace(model.get());
+template <class T>
+void ModelLoad(Argument<std::string> path,
+               Result<RCReference<Model>> result_model,
+               KernelErrorHandler handler, const ExecutionContext &exec_ctx) {
+  auto result = result_model.Allocate();
+  bool work_enqueued = tfrt::EnqueueBlockingWork(
+      exec_ctx.host(), [result = result.CopyRef(), path, handler, exec_ctx]() {
+        auto model = T::load(path.get(), exec_ctx.host());
+        if (auto err = model.takeError()) {
+          result.SetError(err);
+        } else {
+          result.emplace(model.get());
```

Comment:
Would it be better to use `result.emplace(std::move(model.get()))` here?

Reply:
`std::move(...)` changes l-value to r-value, and the return value of any function, like the `model.get()` method here, should have already been an r-value. If this is true, I think we don't need `std::move` here.